backend/src/test/java/heartbeat/service/report/GenerateReporterServiceTest.java

```diff
@@ -1266,6 +1276,126 @@ void shouldPutReportInHandlerWhenCallSaveReporterInHandler() throws IOException
 		verify(asyncReportRequestHandler, times(1)).putReport(reportId, reportResponse);
 	}
 
+	@Test
+	void shouldPutBoardReportIntoHandlerWhenCallGenerateBoardReport() throws IOException, InterruptedException {
+		ObjectMapper mapper = new ObjectMapper();
+		GenerateReportRequest reportRequest = mapper.readValue(new File(REQUEST_FILE_PATH),
+				GenerateReportRequest.class);
+		ReportResponse reportResponse = mapper.readValue(new File(RESPONSE_FILE_PATH), ReportResponse.class);
+		reportRequest.setCsvTimeStamp("1683734399999");
+		MetricsDataReady previousMetricsReady = MetricsDataReady.builder()
+			.isBoardMetricsReady(true)
+			.isPipelineMetricsReady(false)
+			.isSourceControlMetricsReady(false)
+			.build();
+		GenerateReporterService spyGenerateReporterService = spy(generateReporterService);
+
+		doReturn(reportResponse).when(spyGenerateReporterService).generateReporter(reportRequest);
+		when(asyncReportRequestHandler.getMetricsDataReady(reportRequest.getCsvTimeStamp()))
+			.thenReturn(previousMetricsReady);
+
+		spyGenerateReporterService.generateBoardReport(reportRequest);
+
+		Thread.sleep(2000);
+		verify(spyGenerateReporterService, times(1)).generateReporter(reportRequest);
+		verify(spyGenerateReporterService, times(1))
+			.initializeMetricsDataReadyInHandler(reportRequest.getCsvTimeStamp(), reportRequest.getMetrics());
+		verify(spyGenerateReporterService, times(1)).saveReporterInHandler(
+				spyGenerateReporterService.generateReporter(reportRequest), reportRequest.getCsvTimeStamp());
+		verify(spyGenerateReporterService, times(1)).updateMetricsDataReadyInHandler(reportRequest.getCsvTimeStamp(),
+				reportRequest.getMetrics());
+	}
+
+	@Test
+	void shouldPutExceptionInHandlerWhenCallGenerateBoardReportThrowException()
+			throws IOException, InterruptedException {
+		ObjectMapper mapper = new ObjectMapper();
+		GenerateReportRequest reportRequest = mapper.readValue(new File(REQUEST_FILE_PATH),
+				GenerateReportRequest.class);
+		ReportResponse reportResponse = mapper.readValue(new File(RESPONSE_FILE_PATH), ReportResponse.class);
+		reportRequest.setCsvTimeStamp("1683734399999");
+		String boardTimeStamp = "board-1683734399999";
```

Comment:
what is `1683734399999`?
can we refactor with a readable name like `board-20240109232359`, even add milliseconds

must be changed with this comment

Reply:
Done